server/splunk/api.go

```diff
@@ -40,21 +40,21 @@ type AlertActionWHPayload struct {
 type AlertActionFunc func(payload AlertActionWHPayload)
 
 // AddAlertListener registers new listener for alert action
-func (s *splunk) AddAlertListener(channelID string, alertID string, f AlertActionFunc) {
-	s.notifier.addAlertActionFunc(channelID, alertID, f)
+func (s *splunk) AddAlertListener(channelID string, alertID string) error {
+	return s.addAlert(channelID, alertID)
 }
 
-// NotifyAll notifies all listeners about new alert action
-func (s *splunk) NotifyAll(alertID string, payload AlertActionWHPayload) {
-	s.notifier.notifyAll(alertID, payload)
+// Notify notifies all listeners about new alert action
+func (s *splunk) Notify(alertID string, payload AlertActionWHPayload) error {
+	return s.notify(alertID, payload)
 }
 
-func (s *splunk) ListAlert(channelID string) []string {
-	return s.notifier.list(channelID)
+func (s *splunk) ListAlert(channelID string) ([]string, error) {
+	return s.listAlertsInChannel(channelID)
 }
 
 func (s *splunk) DeleteAlert(channelID string, alertID string) error {
-	return s.notifier.delete(channelID, alertID)
+	return s.delete(channelID, alertID)
```

Comment:
What value does this indirection provide?

Reply:
@hanzei  i didn't understood this , do you mean the `s.delete()` function?

Reply:
Yes. Why is there a `(s *splunk) delete(channelID string, alertID string) error ` and a `(s *splunk) DeleteAlert(channelID string, alertID string) error` function?